### ISSUE TITLE
Transcription stops after 2 second silence in the middle of the file

### ISSUE THREAD

**Issue by sarthakagarwal007** [Azure-Samples/cognitive-services-speech-sdk#929]
Hello,
I am using the Speech Cognitive Service and passing an audio file to get the transcription file along with the Pronunciation score. But, I am encountering an issue where the transcription stops whenever there is a short silence.

Is there a config that allows us to process the entire file?

ID in the json: 0ba9da3ec3a34e14be134dde289c8de4

**Comment by glharper** [comment]
@sarthakagarwal007 Thank you for logging this issue. Would you mind sharing your code and the audio file you're using?

**Comment by trrwilson** [comment]
@sarthakagarwal007 we're happy to help diagnose but will need the additional information @glharper requested. It's definitely possible to do what you're looking for and the fix is likely either changing a "RecognizeOnce" call to "StartContinuous" call or modifying a configuration property--that's where having a code snippet would help us advise.

If we don't hear back, we'll close the issue as general procedure. Thanks!

**Comment by sarthakagarwal007** [comment]
Thank you Travis! Yes, can you help me change the configuration of
"RecognizeOnce" with a larger duration? We were successfully able to
implement the "StartContinuous" method, but it gave us the pronunciation
assessment for each individual chunk. And we are looking to have the score
for the entire file at once, instead of us having additional logic (which
would be a non-standard weighted average of the individual chunks.

Hope that makes sense.

Audio file:
https://drive.google.com/file/d/18JUyjgPxyzV6yyan3Ahz49s80ktoG4iK/view?usp=sharing
Below is my Python code:

audio_input = speechsdk.AudioConfig(filename=folder + file_name)
pronunciation_assessment_config =
speechsdk.PronunciationAssessmentConfig(reference_text="Actual Text
spoken in the audio file",

   grading_system=speechsdk.PronunciationAssessmentGradingSystem.HundredMark,

   granularity=speechsdk.PronunciationAssessmentGranularity.Phoneme)
speech_recognizer =
speechsdk.SpeechRecognizer(speech_config=speech_config,
audio_config=audio_input)
pronunciation_assessment_config.apply_to(speech_recognizer)
result = speech_recognizer.recognize_once()

Best,
Sarthak

**Comment by pankopon** [comment]
@sarthakagarwal007 Hi, actually the setting that would enable recognize_once() to process the whole input at once despite moments of silence is not configurable in the Speech SDK anymore - currently it only exists in the service internal configuration. (It used to be configurable by the SDK in the past which is what Travis was thinking about.) So, for the time being you would need to use continuous recognition with some post-processing of results as you mentioned.

However, we do have a work item on the backlog to make this configurable again, tentatively coming this spring. We will update this issue when the work is done and the release schedule can be confirmed.

**Comment by pankopon** [comment]
I just realized that even with the existing work item that would help with occasional silence in input audio, there is also the setting of max duration which is currently 15 seconds for recognize_once() - much longer with continuous recognition - and again, it's service internal only. Your example audio file is 20s so the result would cover only 75% of the file. I have added another work item to the backlog to make the max duration configurable.
